controller/server/main.go

```diff
@@ -183,35 +192,69 @@ func validatePath(path string) (string, error) {
 
 func (s *server) CreateTopology(ctx context.Context, req *cpb.CreateTopologyRequest) (*cpb.CreateTopologyResponse, error) {
 	log.Infof("Received CreateTopology request: %v", req)
-	bp, err := validatePath(req.Topology.Name)
-	if err != nil {
-		return nil, status.Errorf(codes.InvalidArgument, "invalid topology path: %v", err)
+	topoPb := req.GetTopology()
+	if topoPb == nil {
+		return nil, status.Errorf(codes.InvalidArgument, "invalid request: missing topology protobuf")
 	}
-	p := topo.TopologyParams{
-		TopoName:       req.Topology.Name,
-		TopoNewOptions: []topo.Option{topo.WithBasePath(filepath.Dir(bp))},
-		Kubecfg:        req.Kubecfg,
+	if topoPb.GetName() == "" {
+		return nil, status.Errorf(codes.InvalidArgument, "missing topology name")
+	}
+
+	s.muTopo.Lock()
+	defer s.muTopo.Unlock()
+	if _, ok := s.topos[topoPb.GetName()]; ok {
+		return nil, status.Errorf(codes.AlreadyExists, "topology %q already exists", req.Topology.GetName())
+	}
+
+	log.Infof("Validating the topology protobuf...")
```

Comment:
change this log message to indicate that it is only checking the file paths for validity, not the entire proto

Reply:
looking more at this I think the message can just be entirely removed given the follow up "checking config path" logging if it comes up

Reply:
Done